core/entities/Module.php

```diff
@@ -308,7 +308,7 @@ public function saveSetting($setting, $value, $uid = 0, $scope = null)
             return framework\Settings::saveSetting($setting, $value, $this->getName(), $scope, $uid);
         }
 
-        public function deleteSetting($setting, $uid = null, $scope = null)
+        public function deleteSetting($setting, $uid = 0, $scope = null)
```

Comment:
To follow convetion with getSetting() and saveSetting(). If uid is not zero by default then framework\Settings::deleteSetting() does't handle project settings properly where it needs to have uid=0 to delete non-user setting. If uid would be left as null then framework\Settings::deleteSetting() would use current UID instead and not able to delete project specific setting.